dashboard/app/models/contact_rollups_pardot_memory.rb

```diff
@@ -145,11 +158,19 @@ def self.update_pardot_prospects
         old_prospect_data,
         new_contact_data
       )
-      save_sync_results(submissions, errors, Time.now.utc) if submissions.present?
+
+      if submissions.present? && !dry_run
+        save_sync_results(submissions, errors, Time.now.utc)
+      end
     end
 
     submissions, errors = pardot_writer.batch_update_remaining_prospects
-    save_sync_results(submissions, errors, Time.now.utc) if submissions.present?
+
+    if submissions.present? && !dry_run
```

Comment:
would it make sense to move this check into `save_sync_results` so you don't have to check it every time you call `save_sync_results`?

Reply:
Yep, I think makes sense.